roles/system_packages/tasks/main.yml

```diff
@@ -65,14 +65,17 @@
   tags:
     - bootstrap-os
 
-- name: Install packages requirements
+- name: Manage packages (remove and install)
   package:
-    name: "{{ pkgs | dict2items | selectattr('value', 'ansible.builtin.all') | map(attribute='key') }}"
-    state: present
+    name: "{{ item.packages | dict2items | selectattr('value', 'ansible.builtin.all') | map(attribute='key') }}"
+    state: "{{ item.state }}"
   register: pkgs_task_result
   until: pkgs_task_result is succeeded
   retries: "{{ pkg_install_retries }}"
   delay: "{{ retry_stagger | random + 3 }}"
   when: not (ansible_os_family in ["Flatcar", "Flatcar Container Linux by Kinvolk"] or is_fedora_coreos)
+  loop:
+    - { packages: "{{ pkgs_to_remove }}", state: "absent" }
+    - { packages: "{{ pkgs }}", state: "present" }
```

Comment:
If you keep the loop, I think here you need a loop_control: label ( https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#limiting-loop-output-with-label )

Otherwise the whole pkgs structure will be printing, limiting the logs readability substantially. Maybe put remove / install in the label and not in the task name ?